pkg/adaptation/plugin.go

```diff
@@ -249,33 +290,35 @@ func (p *plugin) connect(conn stdnet.Conn) (retErr error) {
 }
 
 // Start Runtime service, wait for plugin to register, then configure it.
-func (p *plugin) start(name, version string) error {
-	var (
-		err     error
-		timeout = getPluginRegistrationTimeout()
-	)
-
-	go func() {
-		err := p.rpcs.Serve(context.Background(), p.rpcl)
-		if err != ttrpc.ErrServerClosed {
-			log.Infof(noCtx, "ttrpc server for plugin %q closed (%v)", p.name(), err)
-		}
-		p.close()
-	}()
+func (p *plugin) start(name, version string) (err error) {
+	if p.wasm == nil {
```

Comment:
```suggestion
	if p.wasm == nil { // skip start for WASM plugins and head right to the registration for events config
```

Reply:
Added, I assume this will also resolve https://github.com/containerd/nri/pull/121#discussion_r1872003996